Added support for webhook parsing

Samuelhertzberg opened the pull request:
Added a file and class "RequestParser.java"

The class contains the method 'parse' which is passed a HttpServletRequest which is parsed to a JSONObject. Then the appropriate fields are extracted.

This should resolve issue #8

Samuelhertzberg committed:
Tests for the request parser

Includes one test sequence which tests two things.

1. The robustness of the parser when passed bad data.

2. The correctness of the parser when passed valid 'normal' data.

Samuelhertzberg committed:
Added support for parsing the github webhook.

The webhook send a HTTP POST request.
The request is parsed to a JSONIbject.
The apporpriate fields are extracted out of the JSON and stored globally 
in the class.
The fields can be accessed via getters.

Samuelhertzberg committed:
Merge branch 'master' of github.com:apeinot/DD2480_lab2_group18 into issue8

apeinot reviewed (changes requested):
Your function and test looks good. They resolve #8 
Can you also parse the following fields that will be usefull for commands:
- repository/clone_url
- repository/name

Add `package CIserver;` at the beginning of your main class file

francislaus commented:
Apart from what @apeinot requested, I think that the implementation and documentation of both the function itself and the test cases looks good.

Jonagoo reviewed (changes requested):
Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.

Aside from this everything looks good.

apeinot commented:
> Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.

I think the clone_url that I asked for is enough (we don't need the ssh_url to clone)

Jonagoo commented:
> > Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.
> 
> I think the clone_url that I asked for is enough (we don't need the ssh_url to clone)

You're probably correct, I've just always went with ssh when cloning so that was the one I recognised.

Samuelhertzberg commented:
Will fix asap

Jonagoo commented:
Another good thing would be to give some sort of return code to see if parse() executed successfully. Just returning a 0 if it's successful and 1 otherwise or something similar would help prevent exceptions later in the code.

Samuelhertzberg committed:
Stored more fields, added error codes

There is now an additional two fields stored clone_url and name.

The parser now returns error codes 1 for successful parsing, 0 for 
connection issues and -1 for parsing issues.

Samuelhertzberg committed:
remove moot code set

Samuelhertzberg committed:
Tests for field extraction/storage method

One test will all bad data. Tests robustness.

One test with partially bad data. Tests robustness and partial data 
read.

One test with all good data. Test complete data read

Samuelhertzberg committed:
Merge branch 'master' of github.com:apeinot/DD2480_lab2_group18 into issue8

Samuelhertzberg committed:
Added comments

Samuelhertzberg commented:
Changed committed and pushed. Error codes specified in parse() doc-comments.

apeinot committed:
Modify call of RequestParser according to the package

This shoud fix the issue in CI for #8

emilelm96 commented on src/main/java/RequestParser.java:
```diff
@@ -0,0 +1,114 @@
+    /**
+    Parse request JSON and get the correct fields
+    @param request The http(s) POST request
+    @return The status code. 1 for success, 0 for connection/response issues and -1 for parsing issues.
```
The parse() function doesn't seem to return -1 when there are parsing issues like specified in the documentation.
